massa-signature/src/signature_impl.rs

```diff
@@ -20,6 +20,22 @@ const PRIVATE_KEY_STRING_PREFIX: &str = "PRI";
 const PUBLIC_KEY_STRING_PREFIX: &str = "PUB";
 const SIGNATURE_STRING_PREFIX: &str = "SIG";
 
+/// `KeyPair` is used for signature and decrypting
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct KeyPair {
+    /// Private key used for signing, wallet dump/load etc...
+    pub private_key: PrivateKey,
+    /// Public key used for verifications
+    pub public_key: PublicKey
+}
+
+impl KeyPair {
+    /// Create a KeyPair from a private key
+    pub fn from_private_key(private_key: PrivateKey) -> KeyPair {
```

Comment:
this should be removed. Everywhere we were using/storing a private key only, we should be using a keypair

Reply:
Done.